src/Engine/SpellFxRenderer.cpp

```diff
@@ -617,23 +617,23 @@ void SpellFxRenderer::_4A7F74(int x, int y, int z) {
 float SpellFxRenderer::_4A806F_get_mass_distortion_value(Actor *pActor) {
     int v2;     // ecx@1
     int v3;     // eax@1
-    double v4;  // st7@2
 
-    v3 = *(int *)&pActor->buffs[ACTOR_BUFF_MASS_DISTORTION]
-              .expireTime.value -
-         pMiscTimer->uTotalTimeElapsed;
+    if (pActor->massDistortionTime == -1)
+        return 1.0;
+
+    assert(pActor->massDistortionTime <= pMiscTimer->uTotalTimeElapsed);
+
+    v3 = 128 - (pMiscTimer->uTotalTimeElapsed - pActor->massDistortionTime);
```

Comment:
Need not to forget to do something with this magic (128 = real world second)

Reply:
True, let me throw in a TODO